src/DsaPrinter.cc

```diff
@@ -644,6 +676,17 @@ struct DsaPrinter : public ModulePass {
         writeGraph(G, Filename);
       }
     } else {
+      if (DsaColorCallSiteSimDot) {
+        auto &CCG = getAnalysis<CompleteCallGraph>();
+        auto &dsaCallGraph = CCG.getCompleteCallGraph();
+
+        m_CG = llvm::make_unique<CallGraphWrapper>(dsaCallGraph);
+
+        m_CG->buildDependencies(); // TODO: this is already done already in
```

Comment:
I don't think you need here `CallGraphWrapper` and therefore the call to `buildDependencies`.
This class basically computes predecessors of CallGraph nodes because the original class doesn't provide that.
In `runOnFunction` you just need to traverse all the callsites defined in `F` and call `colorGraph`.
Instead, you start from `F` and call `m_CG->getUses(F)` which gives the callsite where `F` is called. So you color callsites in other functions where F is called. I think here the order doesn't matter so just go to each callsite defined in `F` and color them.